worlds/sc2/MissionTables.py

```diff
@@ -18,6 +18,28 @@ class MissionPools(IntEnum):
     FINAL = 5
 
 
+class MissionFlags(IntFlag):
+    none        = 0
+    Terran      = 0x1
+    Zerg        = 0x2
+    Protoss     = 0x4
+    NoBuild     = 0x8
+    Defense     = 0x10
+    Timed       = 0x20   # The overall pace of the mission is guided by timers until victory or defeat
+    Kerrigan    = 0x40   # The player controls Kerrigan in the mission
+    VanillaSoa  = 0x80   # The player controls the Spear of Adun in the vanilla version of the mission
+    Nova        = 0x100  # The player controls NCO Nova in the mission
+    AiAlly      = 0x200  # The mission has an AI ally that can be taken over
+    VsTerran    = 0x1000
+    VsZerg      = 0x2000
+    VsProtoss   = 0x4000
+
+    timedDefense = Timed|Defense
+    vsTZ = VsTerran|VsZerg
+    vsTP = VsTerran|VsProtoss
+    vsPZ = VsProtoss|VsZerg
+    vsAll = VsTerran|VsProtoss|VsZerg
```

Comment:
These, together with `none` (do we even need that one?), should be upper-case in my opinion. Technically enum value names should be all-caps even, but all-caps is hard to read so I don't mind breaking that convention. However, people working on this project three years from now shouldn't have to know which specific flags are combinations just to determine the correct casing, especially since they're functionally identical to just using the individual flags.

Reply:
My concern is that they're not functionally the same. If I'm doing a check for timed defense missions, I can't just do `if mission.flags & MissionFlags.timedDefense`, as that's "Timed _or_ defense". The capitalization convention here helps determine if you're working with an alias option or a single bit, and I consider that important.

As to remembering capitalization, for me it's the other way around. The editor provides dropdown options and running will cause a crash if you get the capitalization wrong. Hence the capitalization reminds the programmer of the fact it's a combination, rather than the other way around.

`none` I included as having to specify `MissionFlags(0)` is a little unituitive / breaks mental patterns, and `none` is the 0-value that Berserker has used in several places around AP, such as [here](https://github.com/ArchipelagoMW/Archipelago/pull/3125/files#diff-0f5a189559e017401b555bcac1815941d9c9cbe91169c88ce50818038ab3e44eR25).